eff.go

```diff
@@ -0,0 +1,164 @@
+package eff
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/veandco/go-sdl2/sdl"
+)
+
+const (
+	windowTitle = "Effulgent"
+	frameRate   = 90
+	frameTime   = 1000 / frameRate
+)
+
+// Point container for 2d points
+type Point struct {
+	X int
+	Y int
+}
+
+// Color container for argb colors
+type Color struct {
+	R int
+	G int
+	B int
+	A int
+}
+
+// Drawable interface describing required methods for drawable objects
+type Drawable interface {
+	Init(canvas *Canvas)
+	Draw(canvas *Canvas)
+	Update(canvas *Canvas)
+}
+
+// Canvas creates window and renderer and calls all drawable methods
+type Canvas struct {
+	window    *sdl.Window
+	renderer  *sdl.Renderer
+	drawables []Drawable
+
+	Width  int
+	Height int
+}
+
+// AddDrawable adds a struct that implements the eff.Drawable interface
+func (canvas *Canvas) AddDrawable(drawable Drawable) {
+	canvas.drawables = append(canvas.drawables, drawable)
+}
+
+// Run creates an infinite loop that renders all drawables, init is only call once and draw and update are called once per frame
+func (canvas *Canvas) Run() int {
```

Comment:
This looks awesome! How do you feel about making Canvas an interface instead of a struct then creating an sdl package that has an SDL specific Canvas implementation?